pandas/indexes/base.py

```diff
@@ -432,6 +432,34 @@ def _update_inplace(self, result, **kwargs):
         # guard when called from IndexOpsMixin
         raise TypeError("Index can't be updated inplace")
 
+    def _get_grouper_for_level(self, mapper, level=None):
+        """
+        Get index grouper corresponding to an index level
+
+        Parameters
+        ----------
+        mapper: Group mapping function or None
+            Function mapping index values to groups
+        level : int, default None
+            Index level
+
+        Returns
+        -------
+        grouper : Index
+            Index of values to group on
+        labels : None
+            Array of locations in level_index
+        level_index : None
```

Comment:
this should match the doc-string of the MultiIndex. ( best to use a shared_doc to avoid repeating doc-strings)